vars/createSha512.groovy

```diff
@@ -1,15 +1,26 @@
 Closure call() {
     allowedFileTypes = [".tar.gz", ".zip"]
 
-    return { path -> body: {
-        isAllowed = false
-        for (fileType in allowedFileTypes) {
-            isAllowed |= path.endsWith(fileType)
-        }
-        if (isAllowed) {
-            sh("sha512sum ${path} > ${path}.sha512")
-        } else {
-            echo("Not generating sha for ${path}, doesn't match allowed types ${allowedFileTypes}")
+    return { argsMap -> body: {
+
+        final foundFiles = sh(script: "find $argsMap.artifactPath -type f", returnStdout: true).split()
+
+        for (file in foundFiles) {
+            acceptTypeFound = false
+            for (fileType in allowedFileTypes) {
+                if (file.endsWith(fileType)) {
+                    echo("Creating sha for ${file}")
+                    final sha512 = sh(script: "sha512sum ${file}", returnStdout: true).split()
+                    // Reading the first index of array since sha512sum return [shasum, filename] as output
+                    writeFile file: "${file}.sha512", text: "${sha512[0]}"
+                    acceptTypeFound = true
+                    break
+                }
+            }
+            if (!acceptTypeFound) {
+                echo("Not generating sha for ${argsMap.artifactPath}, doesn't match allowed types ${allowedFileTypes}")
```

Comment:
Message should probably say "none of the files matched ..."

Reply:
The path could be directory or the artifact